pkg/services/ngalert/api/authorization.go

```diff
@@ -1,141 +1,163 @@
 package api
 
 import (
+	"fmt"
+	"net/http"
+
 	"github.com/grafana/grafana/pkg/middleware"
-	"github.com/grafana/grafana/pkg/models"
 	ac "github.com/grafana/grafana/pkg/services/accesscontrol"
+	acmiddleware "github.com/grafana/grafana/pkg/services/accesscontrol/middleware"
+	"github.com/grafana/grafana/pkg/services/dashboards"
+	"github.com/grafana/grafana/pkg/services/datasources"
 	"github.com/grafana/grafana/pkg/web"
 )
 
```

Comment:
I find this file a bit hard to read and I think there's more we can do to tidy it up a bit - we don't have to do it _right_ now but perhaps consider it for the subsequent pull request.

An example of what I'm thinking is to have methods per subsections of alerting and string match on a substring of the path as the first step:

```golang
const grafanaRulerPath = "/api/ruler/grafana/"
const grafanaAlertmanagerPath = "/api/alertmanager/grafana/"

if strings.HasPrefix(path, grafanaRulerPath) {
	resolveForGrafanaRuler(verb, path)
}

if strings.HasPrefix(path, grafanaAlertmanagerPath) {
	resolveForGrafanaAlertmanager(verb, path)
}

// Apply the same for other sections
```

Then, within each function, you can have a map of supported verbs to paths or use the same switch statement you have here. 

As I said, a nit mostly and can def be left for a subsequent PR.